### PR TITLE
Allow variable substitution for whitelisted origins

### PR DESCRIPTION
**Pull request by cpapazian**
Allow for an additional domain (or list of domains) for which variable substitution is allowed in links.

- list of domains is specified via meta tag with name=`amp-link-variable-allowed-origin`
- multiple whitelisted domains can be specified by using whitespace separated list
- allowed variables uses existing whitelist on the element (`data-amp-replace`)

Implements https://github.com/ampproject/amphtml/issues/6141

### CODE REVIEW

File: src/service/url-replacements-impl.js [hunk 1]
```diff
@@ -725,6 +725,34 @@ export class UrlReplacements {
...
+  isAllowedOrigin_(url) {
+    const docInfo = documentInfoForDoc(this.ampdoc);
+
+    if (url.origin == parseUrl(docInfo.canonicalUrl).origin ||
+        url.origin == parseUrl(docInfo.sourceUrl).origin) {
+      return true;
+    }
+
+    const meta = this.ampdoc.getRootNode().querySelector(
+      'meta[name=amp-link-variable-allowed-origin]');
```
Comment (cramforce) [inline src/service/url-replacements-impl.js]: CC @honeybadgerdontcare on meta tag.

File: test/functional/test-url-replacements.js [hunk 2]
```diff
@@ -120,7 +120,16 @@ describes.sandboxed('UrlReplacements', {}, () => {
       },
       document: {
         nodeType: /* document */ 9,
-        querySelector: () => {return {href: canonical};},
+        querySelector: selector => {
+          if (selector.startsWith('meta')) {
+            return {
+              getAttribute: () => {return 'https://whitelisted.com';},
```
Comment (cramforce) [inline test/functional/test-url-replacements.js]: Please add a space separated second origin.

**Comment by cpapazian** [comment]
I signed it!

**Comment by cpapazian** [comment]
@cramforce i've updated the test. are we waiting on anything else?

File: src/service/url-replacements-impl.js (hunk 1 above)
Comment (honeybadgerdontcare) [inline src/service/url-replacements-impl.js]: Thanks, <meta name=amp-...> is whitelisted now.

**Comment by cramforce** [comment]
@rudygalfi We are good with this, right?

**Comment by rudygalfi** [comment]
LGTM

**Pull request merged**